### PR TITLE
Expand username etc. in configured service_account

### LINKED ISSUE #512: {username} does not expand for service_account
Bug description
I am attempting to dynamically assign services accounts to singleuser pods based on the username.

Expected behaviour
c.KubeSpawner.service_account('user-{username}') results in c.KubeSpawner.service_account('user-myusername')

Actual behaviour
serviceaccount "user-{username}" not found"

### PR DESCRIPTION
**Pull request by consideRatio**
Closes #512 by implementing the wanted expansion of the service_account configuration. This change is actually a single line of logic change, but I updated some documentation systematically regarding the expansion of these variables for various configuration options.

### CODE REVIEW

File: kubespawner/spawner.py [hunk 1]
```diff
@@ -1844,7 +1890,7 @@ async def get_pod_manifest(self):
             extra_resource_guarantees=self.extra_resource_guarantees,
             lifecycle_hooks=self.lifecycle_hooks,
             init_containers=self._expand_all(self.init_containers),
-            service_account=self.service_account,
+            service_account=self._expand_all(self.service_account),
```
Comment (consideRatio) [inline kubespawner/spawner.py:1893]: This is the actual change.

File: kubespawner/spawner.py [hunk 2]
```diff
@@ -1844,7 +1894,7 @@ async def get_pod_manifest(self):
...
```
Comment (consideRatio) [inline kubespawner/spawner.py:1897]: This is the actual change, the rest is documentation and tests.

**Comment by yuvipanda** [comment]
Thanks a lot, @consideRatio! We should systematically expand it in everything - how do you think we should approach that?

**Comment by consideRatio** [comment]
@yuvipanda oh I'd say the strategy would be to consider if we could do a single `_expand_all()` call on a high level object rather than individual pieces we add to the big pod object for example. But, if we do that, I see it as a quite big risk that it may need to be reverted because we didn't consider something unknown so I'd be a bit hesitant merging such change without thinking it through a lot.

**Comment by consideRatio** [comment]
Thanks for review and merge @yuvipanda!!! :heart:

**Comment by yuvipanda** [comment]
> But, if we do that, I see it as a quite big risk that it may need to be reverted because we didn't consider something unknown so I'd be a bit hesitant merging such change without thinking it through a lot.

Yeah, I agree - especially since shell scripts might have things inside `{}` all the time.

Maybe we can do this in two steps:

- [ ] Evaluate places that aren't going to contain shell scripts (like serviceaccount) and expand them automatically. I don't think this needs a major version bump.
- [ ] During next major version bump, let's make it expand everything. Folks will then have to explicitly escape any `{}` in shell scripts.

How does that sound?

**Pull request merged**